covid_act_now/delphi_covid_act_now/geo.py

```diff
@@ -0,0 +1,94 @@
+"""Geo-aggregation related functions."""
+
+import numpy as np
+import pandas as pd
+
+from delphi_utils import GeoMapper
+
+from .constants import GEO_RESOLUTIONS
+
+gmpr = GeoMapper()
+
+def positivity_rate(x):
+    """
+    Find Positivity Rate from binomial counts.
+
+    Parameters
+    ----------
+    x: pd.DataFrame
+        Columns: pcr_tests_positive, sample_size, ...
+
+    Returns
+    -------
+    pd.Series
+        Positivity Rate of PCR-specimen tests.
+    """
+    p = x.pcr_tests_positive / x.sample_size
+
+    # To stay consistent with CAN's / CDC's reporting
+    # They use report a 0% rate even if 0 total tests were performed
+    return np.where(x.sample_size == 0, 0, p)
+
+def std_err(x):
+    """
+    Find Standard Error of a binomial proportion.
+
+    Parameters
+    ----------
+    x: pd.DataFrame
+        Columns: val, sample_size, ...
+
+    Returns
+    -------
+    pd.Series
+        Standard error of the positivity rate of PCR-specimen tests.
+    """
+    p = x.val
+    n = x.sample_size
+    return np.sqrt(p * (1 - p) / n)
```

Comment:
Won't this generate `inf` when sample_size is zero?

We've been using a Jeffreys prior for sampling-based indicators with occasional 0 sample sizes. The rule is you add a pseudocount of 1/2 to each possible outcome. This bumps `n` up to 1, so you're no longer dividing by zero.

[Here's an example in the changehc indicator](https://github.com/cmu-delphi/covidcast-indicators/blob/127c97643d3ee6ad0441f039131653d092249284/changehc/delphi_changehc/sensor.py#L133)

You'd apply the jeffrey's prior to the statistic and to the standard error.

Reply:
Yeah the `sample_size` 0 is quite a common issue! This has been producing NaN for the `se` in the current form I think because of the `sqrt`.

Especially for the most recent dates, many counties have a sample_size 0 with 0% positivity rate to testing data not being known yet.

I was thinking of using Jeffreys prior, but was unsure if we are fine with the positivity rate being 0.5 for such cases. Should I go ahead and use Jeffrey's prior anyway?

Reply:
we should run it by Chris and a stats advisor to make sure, but I think so -- 0.5 with a large standard error is what you'd expect if you were guessing without any data to go off of.

Reply:
alternately we could suppress reporting for 0 sample size

Reply:
Hey @tildechris thoughts? CAN publishes records with 0 sample size, which is not ideal from a statistical standpoint. How should we handle that?

* suppress the zeros
* mirror the zeros (which will be awkward to validate)
* use Statistics™️ on the zeros (which will make them 0.5 with a hefty error bar)

Reply:
Hi @krivard, I met with @eujing on this topic and it seems that the least surprising thing to do would be to simply drop entries which are 0 sample size.

If 0 sample size means that there was a lack of observation as opposed to knowing that 0 tests were performed, I think this is also semantically correct.  Even in the case where there is an observation which is 0 ("we didn't do any tests" vs "i don't yet know how many tests we did"), since we're ultimately trying to get to percent positivity, I think that dropping is still the correct thing to do.

Reply:
Sounds good to me -- thanks!